rdflib/plugins/sparql/operators.py

```diff
@@ -613,8 +614,9 @@ def decorator(func):
 
 def unregister_custom_function(uri, func):
```

Comment:
One thing that maybe bothers me a bit, what if you just want to unregister a custom function regardless of the action function identity. Maybe if func is none the check for function identity should not be performed.

Reply:
We still have to check for membership inside the dictionary, else we will get a `KeyError` when deleting a non-existent function. Maybe something like this would work?

```python
def unregister_custom_function(uri, func):
    if func is not None and _CUSTOM_FUNCTIONS.get(uri, (None, None))[0] != func:
        warnings.warn("This function is not registered as %s" % uri.n3())
    elif uri in _CUSTOM_FUNCTIONS:
        del _CUSTOM_FUNCTIONS[uri]
```

Reply:
yes, but lets wait for @nicholascar to give some input, I tagged him for review. I still the the change as is makes sense, but if we are going to change the function we can maybe go a bit further, even if not it is good to discuss some options.

Reply:
I agree, and thanks for the perspective. I've put your simplified implementation in my PR and adjust the tests to match the new behavior